src/components/hooks/usePerson.js

```diff
@@ -18,7 +18,7 @@ export default function usePerson(personID, adapter) {
     return () => {
       subscription.unsubscribe();
     };
-  }, [person]);
+  }, [adapter, person, personID]);
```

Comment:
If you look at what the effect is doing, it says "[run this function after every render](https://reactjs.org/docs/hooks-reference.html#useeffect)", but there is a way to say "well, don't run this after EVERY render, only if after rendering this value has changed".

In most cases, the value changes because of a one-time activity, e.g. you fetch data, but that fetching completes and the value changes. In this case, the data fetching doesn't complete, because we're streaming data. We only need to subscribe once, so we only need to run the effect once. Passing a `[]` assures that the value will never change since empty can't become anything else and so it won't run after any render ever again.
Now, how will the state get updated? We passed the state setter to the subscription, so whenever a new value comes from the stream it will call the setter and that will trigger a re-render

Reply:
That sounds good.